src/api/setting_engine/mod.rs

```diff
@@ -74,6 +74,7 @@ pub struct SettingEngine {
     pub(crate) disable_media_engine_copy: bool,
     pub(crate) srtp_protection_profiles: Vec<SrtpProtectionProfile>,
     pub(crate) receive_mtu: usize,
+    pub(crate) mid_generator: Option<Arc<Box<dyn Fn() -> String + Send + Sync>>>,
```

Comment:
I don't think we need `Box` here? `Arc<dyn Fn() -> Strign + Send + Sync>` ought to be enough.

Reply:
I think let's leave it as sync until someone complains?

Reply:
alright

Reply:
done